packages/vite/src/node/utils.ts

```diff
@@ -1102,19 +1102,26 @@ function normalizeSingleAlias({
 
 /**
  * Transforms transpiled code result where line numbers aren't altered,
- * so we can skip sourcemap generation during dev
+ * so we can skip full sourcemap generation during dev, however we still
+ * have to make a minimal sourcemap with the source property set to the
+ * original file so debuggers know which one to step into.
  */
 export function transformStableResult(
   s: MagicString,
   id: string,
   config: ResolvedConfig
 ): TransformResult {
+  let map
+  if (config.command === 'build') {
+    map = config.build.sourcemap
+      ? s.generateMap({ hires: true, source: id })
+      : null
+  } else {
+    map = new MagicString(s.toString()).generateMap({ hires: true, source: id })
+  }
   return {
     code: s.toString(),
-    map:
-      config.command === 'build' && config.build.sourcemap
-        ? s.generateMap({ hires: true, source: id })
-        : null
+    map
```

Comment:
Thank you for the PR!

Changing this part will affect perf. Calling `.generateMap` is heavy even without `.prepend` or other manipulating methods.
I'm sorry if I mislead you.

Is there any reason you didn't handle this in `transformRequest` as I suggested in the issue?
Rollup merges each plugin's sourcemap, so changing this will result in many sourcemap merge (which is heavy).

Reply:
You are right. I've got a bit lost because I've originally made the fix on v2 and the related code has changed since. Will move it to `transformRequest` if possible. Sorry for wasting your time.

Reply:
@sapphi-red Thanks for the suggestion! Take a look now please. Indeed it is much more simple this way.